pallets/transfer-allowlist/src/benchmarking.rs

```diff
@@ -29,105 +32,282 @@ use super::*;
 
 const BENCHMARK_CURRENCY_ID: FilterCurrency = FilterCurrency::Specific(CurrencyId::ForeignAsset(1));
 
-benchmarks! {
```

Comment:
NOTE: Apart from updating to v2 and adjusting the `where` clause a bit as well as providing the `default_location` wrapper, I did not change anything here apart from removing redundant `clone`s.